### DISCUSSION TITLE
What dependency injection framework do you use with Csla?

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3265] [Polls]
I am interested to know what fellow Csla developers are using for Dependency Injection.

**Comment by rockfordlhotka** [comment]
I'm interested in this as well. I use the built-in capabilities, but have colleagues who really like Autofac a lot.

Within the context of CSLA itself, as you might imagine we only use the capabilities provided by `Microsoft.Extensions.DependencyInjection` - and that's true for the Microsoft technologies as well of course.

**Reply by TheCakeMonster** [reply]
Yes, I always try to make use of the abstractions that the in-built DI framework provides as well. I believe the idea is that you can change out the implementation under the covers if you stick to the abstractions - but I've never found a reason to need to try, so I can't say how successful that would be.

Sticking to the abstractions makes a myriad of frameworks/Nugets work better together, so I would be reluctant to directly rely on features in a particular (alternative) implementation.

**Comment by kcabral817** [comment]
I came from the WPF and used dependency injection with PRISM.  At that time, I was using Unity (another MS ejected tech) that became obsolete and is no longer supported by either MS or the community.  PRISM I feel is going to go the same way.  I have since limited myself to one external tool and that's Automapper.  I use MS's dependency injection because I have no desire to wake up (again) and find out the framework is no longer supported.  It's bad enough that geospatial support is through an addon and not imbedded with .NET.  I think autofac has been around so long that is could be a viable alternative, but I'm too busy to learn another framework and have been burned too many times.

**Reply by Chicagoan2016** [reply]
@kcabral817 I really liked PRISM and have looked at it in the past. I remember Brian Lagunas was supporting Prism.
Is it going away like Unity?

**Reply by rockfordlhotka** [reply]
I think Brian is still maintaining Prism.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I was looking at their repository and there is activity going on 😊

**Reply by Inictus** [reply]
Just to comment on this, Prism doesn't do dependency injection itself. It currently supports Unity and DryIoc. Personally I use Prism with Unity even knowing Unity is sort of in limbo. I'm just too lazy to learn DryIoc right now. Too many other things to learn at this point lol.

**Comment by ajohnstone-ks** [comment]
We are currently using Unity, but given that's no longer supported we'll eventually move to the MS DI extensions.

**Comment by ctwoodwa** [comment]
I use out-of-the-box solutions as they are supported longer and tend to incorporate the useful features of the trendy packages eventually.  I started looking into AutoFac because of a desire to build a plugin/modular-based solution.  I recently found a project called [Octane](https://www.oqtane.org/) that is already plugin/module-based. Octane looks to be created by the same people who created DotNetNuke from years past.
